fix(dropdown): Updates dropdown icon, background, and shadow

srambach opened the pull request:
Updates background color and box shadow of menu and changes dropdown indicator to be a caret icon instead of angle thing.

Fixes #455

patternfly-build commented:
Deploy preview for *pf-next* ready!

Built with commit e6303066f2e093497ae32ebaea984d2c413cf50f

https://deploy-preview-481--pf-next.netlify.com

thomasmaas commented:
Seems it's a bit off compared to the visual design:

Visual design:
<img width="48" alt="screenshot 2018-07-19 11 12 23" src="https://user-images.githubusercontent.com/54224/42933316-c84e791a-8b44-11e8-99cb-e6a1bff53004.png">

HTML + CSS:
<img width="59" alt="screenshot 2018-07-19 11 06 49" src="https://user-images.githubusercontent.com/54224/42933325-cb696a1a-8b44-11e8-948d-b0343c0bdde5.png">

Visual design:
<img width="38" alt="screenshot 2018-07-19 11 12 35" src="https://user-images.githubusercontent.com/54224/42933317-c86ea136-8b44-11e8-951b-3888b108e4c5.png">

HTML + CSS:
<img width="39" alt="screenshot 2018-07-19 11 06 56" src="https://user-images.githubusercontent.com/54224/42933326-cb85b3aa-8b44-11e8-8c80-ae17a8982216.png">

jgiardino commented on src/patternfly/components/Dropdown/examples/dropdown-dropup-example.hbs:
```diff
@@ -1,2 +1,8 @@
-{{#> dropdown id="dropdown-example-dropup-expanded" pf-c-dropdown--IsExpanded="true" pf-c-dropdown--Modifier="pf-m-dropup"}}Dropup{{/dropdown}}
+{{#> dropdown id="dropdown-example-dropup-collapsed" pf-c-dropdown--Modifier="pf-m-dropup"}}
+  Dropup
+    <i class="fas fa-caret-down pf-c-dropdown__toggle-icon"></i> 
```
Should this be defined in a handlebar so that we have one place where it's defined?

`<i class="fas fa-caret-down pf-c-dropdown__toggle-icon"></i>`

srambach commented:
@kybaker I added an -xs spacer between the dropdown field and the menu, and changed the background-clip to try to extend the selection highlight that last pixel through the transparent border (which cannot be removed because it's needed for the high-contrast mode). The box-shadow is the md one as specified in the issue, so if that's not correct, please let me know which it should be (and anything else.) Thanks!

srambach commented:
@kybaker I tweaked the shadow. See what you think.

kybaker commented:
@srambach Looks great!

michael-coker commented:
Looks like the `.pf-c-dropdown__menu` is no longer `position: absolute;` and the `.pf-m-dropup` demo isn't working properly. I'm assuming `position: relative;` would also cause the menu to push around content below it when the dropdown is expanded?

jgiardino commented:
These updates look great!

kybaker commented:
@srambach Spacing looks good.
